fiona/ogrext.pyx

```diff
@@ -480,38 +477,37 @@ cdef class Session:
         else:
             drivers = None
 
+        if 'encoding' in kwargs:
+            if not kwargs['encoding']:
+                kwargs.pop('encoding')
+            else:
+                kwargs['encoding'] = kwargs['encoding'].upper()
+
```

Comment:
I think this can be simplified to only one dictionary lookup:

```python
encoding = kwargs.pop("encoding", None)
if encoding:
    kwargs["encoding"] = encoding.upper()
```